doc/developer/guide-changes.md

```diff
@@ -0,0 +1,699 @@
+the complexity can accrete to result in very confusing, unmaintainable code.
+
+Some matters can be more subjective. When you have a preference, but reasonable
+folks can disagree, prefix your comment with "nit:" to indicate that it is not
+a hard requirement for the PR to merge.
+
+### Mentoring
+
+When someone is new to Rust, code review is one of the primary components of
+their feedback loop. Working through the various Rust tutorials and blog posts
+is important, but attempting to put those learnings into practice, and then
+getting feedback from someone more experienced, can really accelerate their
+learning curve.
+
+The same philosophy applies to someone who is experienced with Rust, but new to
+a region of the codebase. Each subsystem has a unique architecture and set of
+tradeoffs to manage, and code review is an important component in communicating
+these particulars to engineers who are entering that region of the codebase for
+the first time. (Ideally these points are also documented somewhere, but even
+when they are, it is unrealistic to expect new engineers to read and absorb all
+of these points before submitting their first PR.) For example, the `sql` crate
+is not particularly allocation-sensitive at the moment, but the `dataflow` has a
+number of hot loops that are extremely allocation sensitive. Engineers
+unfamiliar with the `dataflow` crate will need to be reminded of this in their
+first few PRs.
+
+When reviewing code from a newcomer, spend some extra time rendering feedback on
+these various design points. Make sure to mark comments as "nits" where
+appropriate to avoid overwhelming the reviewee. It is important that newcomers
+are able to make forward progress while they are learning the ropes.
+
+### Review tools
+
+One of the key tenets of code review is that engineers must be permitted to
+be productive. Therefore, they are allowed to choose whichever commit strategy
+they prefer—and you as reviewer must cope.
+
+The flip side of this is that you may use either GitHub reviews or Reviewable as
+the reviewer, whichever you prefer. Beware that with the semantic commit
+strategy, Reviewable works much better, as it can track a set of commits across
+a force push.
+
+### Latency expectations
+
+One of the most frustrating parts of code review arises when reviewers are slow
+to respond. Slow code reviews decrease the velocity of the entire engineering
+organization and breed resentment for the code review process.
+
+As a rule, **PRs should be reviewed as soon as possible after they are
+submitted.**
+
+The exception to this rule is if you are actively in a flow state. Don't
+interrupt your "maker time" to review PRs. But if you are already interrupted,
+e.g. because of a meeting or because of a lunch break, try to clear out your
+review queue when you get back to your desk, before you get back into the flow.
+
+As a hard upper limit, **the time to first response should never exceed one
```

Comment:
that seems pretty low latency. I'm not against this in theory, but I don't think we are meeting this latency bound very often. If this is meant to be aspirational - totally fine and maybe worth rewording? If this is meant as the actual policy - I think it might need to be socialized more?

Reply:
Yeah, proposing this guideline is actually the catalyst for this entire document. I think this is possibly the single most important thing we can do to increase the velocity of the entire engineering org.

Reply:
Can definitely socialize this more before this merges.